x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/stats/StatsHolder.java

```diff
@@ -34,15 +34,30 @@ public void setProgressTracker(List<PhaseProgress> progress) {
         progressTracker = new ProgressTracker(progress);
     }
 
+    /**
+     * Updates the progress tracker with potentially new in-between phases
+     * that were introduced in a later version while making sure progress indicators
+     * are correct.
+     * @param analysisPhases the new analysis phases
```

Comment:
To me the phrase "new analysis phases" suggests that `analysisPhases` is a diff compared to some previous state whereas IIUC this is the full new list of phases.
Do you think it could be rephrased?

Reply:
sure